lib/internal/util.js

```diff
@@ -101,13 +104,29 @@ function deprecate(fn, msg, code) {
   return deprecated;
 }
 
-function decorateErrorStack(err) {
+function decorateErrorStack(err, options) {
   if (!(isError(err) && err.stack) ||
       getHiddenValue(err, kDecoratedPrivateSymbolIndex) === true)
     return;
 
   const arrow = getHiddenValue(err, kArrowMessagePrivateSymbolIndex);
 
+  if (options.format === 'commonjs' && arrow) {
+    const {
+      childSpecifier
+    } = options;
+    const importStatement = arrow.split('\n')[1];
+    const namedImports = StringPrototypeMatch(importStatement, /{.*}/)[0];
+    err.message = `The requested module '${childSpecifier}' is expected ` +
+      'to be of type CommonJS, which does not support named exports. ' +
+      'CommonJS modules can be imported by importing the default export. ' +
+      `The following code will work: const ${namedImports} = ` +
+      `require('${childSpecifier}');`;
```

Comment:
But `require` doesn’t work in ESM, does it?

Reply:
fixed, doing too many things at once LOL